### PR TITLE
Scenario 7.1.1 , 5.1.1

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
[OKTA-397543](https://oktainc.atlassian.net/browse/OKTA-397543)
```
  Scenario 7.1.1: Mary Logs in with Social IDP
	GIVEN Mary navigates to Login with Social IDP
	WHEN she clicks the "Login with Google" button in the embedded Sign In Widget
	AND logs in to Google
	THEN she is redirected back to the Sample App
	// AND the Widget is reinitialized to re-enter the remediation flow to complete authentication
	// AND Mary is redirected back to the Root View
```

[OKTA-397423](https://oktainc.atlassian.net/browse/OKTA-397423)
```
  Scenario 5.1.1: Mary Logs in with Social IDP
	GIVEN Mary navigates to the Login View
	WHEN she clicks the "Login with Google" button
	AND logs in to Google
	THEN she is redirected to the Root View
    AND an application session is created
```

### CODE REVIEW

File: samples/test/support/check/checkURLPath.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default async (falseCase: boolean, expectedPath: string) => {
      * @type {String}
      */
     let currentUrl = await browser.getUrl();
-    currentUrl = currentUrl.replace(/http(s?):\/\//, '');
+    currentUrl = currentUrl.replace(/http(s?):\/\//, '').replace(/#(.*?)$/, '');
```
Comment (denysoblohin-okta) [inline samples/test/support/check/checkURLPath.ts:13]: Facebook redirects to `/#_=_`

File: samples/test/features/social-idp-with-widget.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Feature: Direct Auth with Self Hosted Sign In Widget Social Login with 1 Social IDP
+
+  Background:
+	  Given a SPA, WEB APP or MOBILE Policy that defines Password as the only factor required for authentication
+	  #And a configured IDP object for Google
```
Comment (shuowu) [inline samples/test/features/social-idp-with-widget.feature:5]: nit: Google -> Facebook

File: samples/test/features/social-idp.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
+Feature: Direct Auth with Self Hosted Sign In Widget Social Login with 1 Social IDP
+
+Background:
+	Given a SPA, WEB APP or MOBILE Policy that defines Password as the only factor required for authentication
+    # And a configured IDP object for Google
```
Comment (shuowu) [inline samples/test/features/social-idp.feature:5]: nit: Google -> Facebook

File: samples/test/support/action/clickLoginWithFacebookInWidget.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import clickElement from './clickElement';
+import { OktaSignInV1, OktaSignInOIE } from  '../selectors';
+
+export default async () => {
+  const OktaSignIn = process.env.ORG_OIE_ENABLED ? OktaSignInOIE : OktaSignInV1;
```
Comment (shuowu) [inline samples/test/support/action/clickLoginWithFacebookInWidget.ts:5]: Looks like this logic appears in lots of places, how about extract to a util function something like `getOktaSignInForm`?

File: samples/test/support/action/signInIntoFacebook.ts [hunk 5]
```diff
@@ -1,11 +1,29 @@
 import { getConfig } from '../../util';
-import FacebookSignIn from '../selectors/FacebookSignIn';
-import clickElement from './clickElement';
+import FacebookSignIn from  '../selectors/FacebookSignIn';
```
Comment (shuowu) [inline samples/test/support/action/signInIntoFacebook.ts:2]: nit: extra space, sorry I am doing the linting work

File: samples/test/support/check/checkURLPath.ts (hunk 1 above)
Comment (shuowu) [reply samples/test/support/check/checkURLPath.ts:13]: Can you explain more about what happens here? My understanding is the app redirect to facebook for SSO, then facebook redirect to okta, then okta redirects to the app.

File: samples/test/support/wait/waitForDisplayed.ts [hunk 6]
```diff
@@ -4,20 +4,22 @@ import type { Selector } from 'webdriverio';
...
-    /**
-     * Maximum number of milliseconds to wait for
-     * @type {Int}
-     */
-    const ms = 10000;
-
-    const el = await $(selector);
-    await el.waitForDisplayed({
+export default async (selector: Selector, falseCase: any = false, ms = 10000) => {
+    // Unlike `el.waitForDisplayed` approach, this way we can catch timeout exception
```
[the remark this answers is not in the document]
Comment (shuowu) [reply samples/test/support/wait/waitForDisplayed.ts:12]: Do you still see this exception with cookie being cleared from facebook domain?

File: samples/test/support/check/checkURLPath.ts (hunk 1 above)
Comment (denysoblohin-okta) [reply samples/test/support/check/checkURLPath.ts:13]: Yes, your understanding is correct.
Facebook redirects to `https://javascript-idx-sdk.okta.com/oauth2/v1/authorize/callback?code=AAA#_=_`.
Which redirects to `http://localhost:8080/login/callback?interaction_code=AAA&state=AAA`
Then I see home page with URL `http://localhost:8080/#_=_`.

File: samples/test/support/wait/waitForDisplayed.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (denysoblohin-okta) [reply samples/test/support/wait/waitForDisplayed.ts]: I wanted to automatically bypass this window:
Screenshot 2021-06-17 at 22 50 46
It does not depend from cookie.
I've removed code which was responsible for this.

File: samples/test/support/check/checkURLPath.ts (hunk 1 above)
Comment (shuowu) [reply samples/test/support/check/checkURLPath.ts:13]: I see, one idea is to make checkURLPath accept a `transformUrl` function, then we can customize the expected url based on different scenarios, so some specially handling won't affect the general cases.

**Review by shuowu — APPROVED** [review]

File: samples/test/support/action/clickLoginWithFacebookInWidget.ts (hunk 4 above)
Comment (denysoblohin-okta) [reply samples/test/support/action/clickLoginWithFacebookInWidget.ts]: Done

File: samples/test/support/check/checkURLPath.ts (hunk 1 above)
Comment (denysoblohin-okta) [reply samples/test/support/check/checkURLPath.ts]: I've added 3rd param `removeHash` (default false) to make it simpler, if you don't mind

**Pull request closed without merging**